samples/nrf9160/modem_shell/src/location/location_shell.c

```diff
@@ -0,0 +1,401 @@
+};
+
+struct location_shell_cmd_args {
+	enum location_shell_command command;
+};
+
+/******************************************************************************/
+static const char location_usage_str[] =
+	"Usage: location <subcommand> [options]\n"
+	"\n"
+	"Subcommands:\n"
+	"  get:      Requests the current position or starts periodic position updates.\n"
+	"  cancel:   Cancel/Stop on going request. No options\n";
+
+static const char location_get_usage_str[] =
+	"Usage: location get [--method <method>] [--interval <secs>]\n"
+	"[--gnss_accuracy <acc>] [--gnss_timeout <timeout in secs>]\n"
+	"[--cellular_timeout <timeout in secs>] [--cellular_service <service_string>]\n"
+	"[--wifi_timeout <timeout in secs>] [--wifi_service <service_string>]\n"
+	"Options:\n"
+	"  --method,           Location method: 'gnss', 'cellular' or 'wifi'. Multiple\n"
+	"                      '--method' parameters may be given to indicate list of\n"
+	"                      methods in priority order.\n"
+	"  --interval,         Position update interval in seconds\n"
+	"                      (default: 0 = single position)\n"
+	"  --gnss_accuracy,    Used GNSS accuracy: 'low' or 'normal'\n"
+	"  --gnss_timeout,     GNSS timeout in seconds\n"
+	"  --cellular_timeout, Cellular timeout in seconds\n"
+	"  --cellular_service, Used cellular positioning service:\n"
+	"                      'any' (default), 'nrf', 'skyhook', 'here' or 'polte'\n"
+	"  --wifi_timeout,     WiFi timeout in seconds\n"
+	"  --wifi_service,     Used WiFi positioning service:\n"
+	"                      'any' (default), 'nrf', 'skyhook' or 'here'\n";
+
+/******************************************************************************/
+
+/* Following are not having short options */
+enum {  LOCATION_SHELL_OPT_INTERVAL         = 1001,
+	LOCATION_SHELL_OPT_GNSS_ACCURACY    = 1002,
+	LOCATION_SHELL_OPT_GNSS_TIMEOUT     = 1003,
+	LOCATION_SHELL_OPT_CELLULAR_TIMEOUT = 1004,
+	LOCATION_SHELL_OPT_CELLULAR_SERVICE = 1005,
+	LOCATION_SHELL_OPT_WIFI_TIMEOUT     = 1006,
+	LOCATION_SHELL_OPT_WIFI_SERVICE     = 1007,
+};
+
+/* Specifying the expected options */
+static struct option long_options[] = {
+	{ "method", required_argument, 0, 'm' },
+	{ "interval", required_argument, 0, LOCATION_SHELL_OPT_INTERVAL },
+	{ "gnss_accuracy", required_argument, 0, LOCATION_SHELL_OPT_GNSS_ACCURACY },
+	{ "gnss_timeout", required_argument, 0, LOCATION_SHELL_OPT_GNSS_TIMEOUT },
+	{ "cellular_timeout", required_argument, 0, LOCATION_SHELL_OPT_CELLULAR_TIMEOUT },
+	{ "cellular_service", required_argument, 0, LOCATION_SHELL_OPT_CELLULAR_SERVICE },
+	{ "wifi_timeout", required_argument, 0, LOCATION_SHELL_OPT_WIFI_TIMEOUT },
+	{ "wifi_service", required_argument, 0, LOCATION_SHELL_OPT_WIFI_SERVICE },
+	{ 0, 0, 0, 0 }
+};
+
+/******************************************************************************/
+
+static void location_shell_print_usage(const struct shell *shell,
+				       struct location_shell_cmd_args *loc_cmd_args)
```

Comment:
Looks like we could pass just` enum location_shell_command command` into this function and actually then we can remove `struct location_shell_cmd_args` completely.